presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/PushAggregatesIntoTableScanRuleSet.java

```diff
@@ -0,0 +1,190 @@
+                    .entrySet().stream()
+                    .collect(toImmutableMap(entry -> entry.getKey().getName(), Map.Entry::getValue));
+            Map<Symbol, AggregationNode.Aggregation> aggregations = node.getAggregations();
+
+            ImmutableList<AggregateFunction> aggregateFunctions = toAggFunction(aggregations);
+
+            if (aggregateFunctions.isEmpty()) {
+                return Result.empty();
+            }
+
+            GroupingSetDescriptor groupingSets = node.getGroupingSets();
+
+            if (groupingSets.getGroupingSetCount() > 1) {
+                // Let the PushAggregationToTableScanWithGroupingSets rule handle this.
+                return Result.empty();
+            }
+
+            List<ColumnHandle> groupByColumns = groupingSets.getGroupingKeys().stream()
+                    .map(assignments::get)
+                    .collect(toList());
+
+            Optional<AggregatePushdownResult<TableHandle>> connectorTableHandleAggregatePushdownResult = metadata.applyAggregation(context.getSession(),
+                    tableScan.getTable(),
+                    aggregateFunctions,
+                    assignments,
+                    Optional.of(ImmutableList.of(groupByColumns)));
+
+            if (!connectorTableHandleAggregatePushdownResult.isPresent()) {
+                return Result.empty();
+            }
+
+            AggregatePushdownResult<TableHandle> aggPushdownResult = connectorTableHandleAggregatePushdownResult.get();
+
+            TableHandle handle = aggPushdownResult.getHandle();
+            ImmutableList.Builder<Symbol> newScanOutputs = new ImmutableList.Builder<>();
+            ImmutableMap.Builder<Symbol, ColumnHandle> newScanAssignments = new ImmutableMap.Builder<>();
+            Map<String, ProjectionApplicationResult.Assignment> nameToAssignment = aggPushdownResult.getAssignments().stream()
+                    .collect(Collectors.toMap(ProjectionApplicationResult.Assignment::getVariable, Function.identity()));
+            for (Symbol aggOutputSymbol : node.getOutputSymbols()) {
+                ProjectionApplicationResult.Assignment assignment = nameToAssignment.get(aggOutputSymbol.getName());
+
+                if (assignment == null) {
+                    //TODO We pushdown all or nothing for V1 but we need to extend this in future.
+                    return Result.empty();
+                }
+                newScanOutputs.add(aggOutputSymbol);
+                newScanAssignments.put(aggOutputSymbol, assignment.getColumn());
+            }
+
+            return Result.ofPlanNode(
+                    TableScanNode.newInstance(
+                            tableScan.getId(),
+                            handle,
+                            newScanOutputs.build(),
+                            newScanAssignments.build()));
+        }
+    }
+
+    private ImmutableList<AggregateFunction> toAggFunction(Map<Symbol, AggregationNode.Aggregation> aggregations)
```

Comment:
Use interface type for collection: `List<AggregateFunction>`. Also, don't abbreviate names: `toAggregateFunction` would be better.

Reply:
that was my bad, should have noticed the type difference. name changed.